src/pages/workspace/tags/WorkspaceTagsPage.tsx

```diff
@@ -106,11 +143,28 @@ function WorkspaceTagsPage({policyTags, route}: WorkspaceTagsPageProps) {
                         icon={Illustrations.Tag}
                         title={translate('workspace.common.tags')}
                         shouldShowBackButton={isSmallScreenWidth}
-                    />
-                    <View style={[styles.ph5, styles.pb5, styles.pt3]}>
+                    >
+                        {!isSmallScreenWidth && policyTags && settingsButton}
+                    </HeaderWithBackButton>
+                    {isSmallScreenWidth && policyTags && <View style={[styles.pl5, styles.pr5]}>{settingsButton}</View>}
+                    <View style={[styles.ph5, styles.pb5]}>
                         <Text style={[styles.textNormal, styles.colorMuted]}>{translate('workspace.tags.subtitle')}</Text>
                     </View>
-                    {tagList.length ? (
+                    {isLoading && (
+                        <ActivityIndicator
+                            size={CONST.ACTIVITY_INDICATOR_SIZE.LARGE}
+                            style={[styles.flex1]}
+                            color={theme.spinner}
+                        />
+                    )}
+                    {tagList.length === 0 && !isLoading && (
```

Comment:
This line caused a [bug](https://github.com/Expensify/App/issues/42220#issuecomment-2124305407) later on. The deleted tags in offline mode had there state set as `pending`.